pylana/logs.py

```diff
@@ -187,16 +188,16 @@ def request_event_csv(self, log_id: str, **kwargs) -> Response:
             'graphControl': {'sizeControl': 'Frequency', 'colorControl': 'AverageDuration'}})
         return self.get(f'/api/eventCsvWithFilter?request={request_field}', **kwargs)
 
-    def get_event_log(self, log_name: str = None, log_id: str = None, **kwargs) -> pd.DataFrame:
+    def get_event_log(self, log_name: str = None, log_id: str = None,
+                      mining_request: Optional[dict] = None, **kwargs) -> pd.DataFrame:
         """
         get the enriched event log as a pandas dataframe
 
         only columns with time stamps are type cast, the other columns remain objects
         """
-
         log_id = log_id or self.get_log_id(log_name)
-        resp = self.request_event_csv(log_id)
-        csv_stream = io.StringIO(resp.text)
+        resp = self.request_event_csv(log_id, mining_request, **kwargs)
+        csv_stream = io.BytesIO(resp.content)
         return pd.read_csv(csv_stream, dtype='object', **kwargs)
```

Comment:
I just realised, that the `kwargs` make no sense here for the data frame. We always know in which form we receive the csv, since it't our api that sends it. If someone knows about the kwargs, he or she will be able to read in the payload of the response anyway. So ot would be simpler (and more consistent with the other methods) to leave the `kwargs` from the `pd.read_csv`.

Reply:
If we request a non-existing log, the result of this function is an empty data frame with one columns named `Bad Request`, which is the requests' response content,

```python
Empty DataFrame
Columns: [Bad Request]
Index: []
```

The result should rather be n empty dataframe. One way to do this would be to use a construct like

```python
if resp.status_code >= 400:
    return pd.DataFrame()
```

after line 199. A `try ... except` with `resp.raise_for_status()` would also be possible.